server/sonar-server/src/main/java/org/sonar/server/es/IndexerStartupTask.java

```diff
@@ -54,16 +62,56 @@ private boolean indexesAreEnabled() {
   }
 
   private void indexEmptyTypes(StartupIndexer indexer) {
-    Set<IndexType> emptyTypes = getEmptyTypes(indexer);
-    if (!emptyTypes.isEmpty()) {
-      log(indexer, emptyTypes, "...");
-      indexer.indexOnStartup(emptyTypes);
-      log(indexer, emptyTypes, "done");
+    Set<IndexType> uninizializedTypes = getUninitializedTypes(indexer);
+    if (!uninizializedTypes.isEmpty()) {
+      log(indexer, uninizializedTypes, "...");
+      indexer.indexOnStartup(uninizializedTypes);
+      uninizializedTypes.forEach(this::setInitialized);
+      log(indexer, uninizializedTypes, "done");
     }
   }
 
-  private Set<IndexType> getEmptyTypes(StartupIndexer indexer) {
-    return indexer.getIndexTypes().stream().filter(esClient::isEmpty).collect(toSet());
+  private Set<IndexType> getUninitializedTypes(StartupIndexer indexer) {
+    return indexer.getIndexTypes().stream().filter(this::getUninitialized).collect(toSet());
+  }
+
+  private boolean getUninitialized(IndexType indexType) {
+    String setting = esClient.nativeClient().admin().indices().prepareGetSettings(indexType.getIndex()).get().getSetting(indexType.getIndex(),
+      getInitializedSettingName(indexType));
+	return !"true".equals(setting);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)